packages/front-end/pages/settings/team/[tid].tsx

```diff
@@ -0,0 +1,224 @@
+import router from "next/router";
+import { FC, useState } from "react";
+import { TeamInterface } from "back-end/types/team";
+import { datetime } from "shared/dates";
+import Link from "next/link";
+import { MemberRoleWithProjects } from "back-end/types/organization";
+import { FaUserLock } from "react-icons/fa";
+import { useForm } from "react-hook-form";
+import { useAuth } from "@/services/auth";
+import LoadingOverlay from "@/components/LoadingOverlay";
+import usePermissions from "@/hooks/usePermissions";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import { GBAddCircle, GBCircleArrowLeft, GBEdit } from "@/components/Icons";
+import RoleSelector from "@/components/Settings/Team/RoleSelector";
+import TeamModal from "@/components/Teams/TeamModal";
+import useApi from "@/hooks/useApi";
+import Modal from "@/components/Modal";
+import { AddMembersModal } from "@/components/Teams/AddMembersModal";
+
+const TeamPage: FC = () => {
+  const { apiCall } = useAuth();
+  const { tid } = router.query as { tid: string };
+  const [teamModalOpen, setTeamModalOpen] = useState<boolean>(false);
+  const [permissionModalOpen, setPermissionModalOpen] = useState<boolean>(
+    false
+  );
+  const [memberModalOpen, setMemberModalOpen] = useState<boolean>(false);
+
+  const permissions = usePermissions();
+  const canManageTeam = permissions.check("manageTeam");
+
+  const { data, mutate } = useApi<{
+    team: TeamInterface;
+  }>(`/teams/${tid}`);
+
+  if (!data) {
+    return <LoadingOverlay />;
+  }
+
+  const { team } = data;
+  const isEditable = !team.managedByIdp;
+
+  const PermissionsModal = () => {
```

Comment:
Don't define a React component instead of another component

Reply:
Moved to `components/Settings/Teams``